src/js/models/layer.js

```diff
@@ -544,7 +567,19 @@ define(function (require, exports, module) {
      * @return {boolean}  
      */
     Layer.prototype.hasLayerEffect = function () {
-        return !this.innerShadows.isEmpty() || !this.dropShadows.isEmpty();
+        return !!this.effects.find(function (effects) { return !effects.isEmpty(); });
+    };
+    
+    /**
+     * True if the layer has unsupported layer effect.
+     * @return {boolean}  
+     */
+    Layer.prototype.hasUnsupportedLayerEffect = function () {
```

Comment:
Since this has an `O(n)` lookup, it might be worthwhile to make this a cached property instead of a method. See, e.g., `unsupported` and `object.getCachedSpecs` above in this file. (I would be happy to explain how that works offline, btw.)